source/brailleDisplayDrivers/seikantk.py

```diff
@@ -0,0 +1,303 @@
+	0x000200: "RJ_CENTER",
+	0x000400: "RJ_LEFT",
+	0x000800: "RJ_RIGHT",
+	0x001000: "RJ_UP",
+	0x002000: "RJ_DOWN"
+}
+
+SEIKA_REQUEST_INFO = b"\x03\xff\xff\xa1"
+SEIKA_INFO = b"\xff\xff\xa2"
+SEIKA_SEND_TEXT = b"\x2c\xff\xff\xa3"
+SEIKA_ROUTING = b"\xff\xff\xa4"
+SEIKA_KEYS = b"\xff\xff\xa6"
+SEIKA_KEYS_ROU = b"\xff\xff\xa8"
+
+SEIKA_CONFIG = b"\x50\x00\x00\x25\x80\x00\x00\x03\x00"
+SEIKA_CMD_ON = b"\x41\x01"
+
+vidpid = "VID_10C4&PID_EA80"
+hidvidpid = "HID\\VID_10C4&PID_EA80"
+SEIKA_NAME = "seikantk"
+
+
+def _getDotNames():
+	dotNames = {}
+	for dotNum in range(1, 9):
+		keyName = globals()[f"DOT_{dotNum}"]
+		dotNames[keyName] = f"d{dotNum}"
+	return dotNames
+
+
+_dotNames = _getDotNames()
+bdDetect.addUsbDevices(SEIKA_NAME, bdDetect.KEY_HID, {vidpid, })
+
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	_dev: hwIo.IoBase
+	name = SEIKA_NAME
+	# Translators: Name of a braille display.
+	description = _("Seika Notetaker")
+	path = ""
+	isThreadSafe = True
+	for d in hwPortUtils.listHidDevices():
+		if d["hardwareID"].startswith(hidvidpid):
+			path = d["devicePath"]
+
+	@classmethod
+	def check(cls):
+		return True
+
+	@classmethod
+	def getManualPorts(cls):
+		return cls.path
+
+	def __init__(self, port="hid"):
+		super().__init__()
+		self.numCells = 0
+		self.numBtns = 0
+		self.handle = None
+
+		self._hidBuffer = b""
+		self._command: typing.Optional[bytes] = None
+		self._argsLen: typing.Optional[int] = None
+		log.info(f"Seika Notetaker braille driver path: {self.path}")
+
+		if self.path == "":
+			raise RuntimeError("No MINI-SEIKA display found, no path found")
+		self._dev = dev = hwIo.Hid(path=self.path, onReceive=self._onReceive)
+		if dev._file == INVALID_HANDLE_VALUE:
+			raise RuntimeError("No MINI-SEIKA display found, open error")
+		dev.setFeature(SEIKA_CONFIG)  # baud rate, stop bit usw
+		dev.setFeature(SEIKA_CMD_ON)  # device on
+		dev.write(SEIKA_REQUEST_INFO)  # Request the Info from the device
+
+		# wait and try to get info from the Braille display
+		for i in range(MAX_READ_ATTEMPTS):  # the info-block is about
+			dev.waitForRead(READ_TIMEOUT_SECS)
+			if self.numCells:
+				log.info(
+					f"Seika notetaker on USB-HID,"
+					f" Cells {self.numCells}"
+					f" Buttons {self.numBtns}"
+				)
+				break
+
+		if self.numCells == 0:
+			dev.close()
+			raise RuntimeError("No MINI-SEIKA display found, no response")
+
+	def terminate(self):
+		try:
+			super().terminate()
+		finally:
+			self._dev.close()
+
+	def display(self, cells: List[int]):
+		# cells will already be padded up to numCells.
+		cellBytes = SEIKA_SEND_TEXT + self.numCells.to_bytes(1, 'little') + bytes(cells)
+		self._dev.write(cellBytes)
+
+	def _onReceive(self, data: bytes):
+		"""
+		Note: Further insight into this function would be greatly appreciated.
+		- on each call, read three bytes only the second byte is used.
+		- first 3 bytes: command
+		- 1 byte: specify total length in bytes?
+		- variable length: arguments for command type
```

Comment:
I found the first bullet point a bit hard to understand, so I rewrote this using my understanding of this function. 

```suggestion
		Note: Further insight into this function would be greatly appreciated.
		This function is a state machine which maintains a buffer to read and process a command.
		Each call reads three bytes, however only the second byte is used in our buffer.
		Using the buffer
		- the first 3 bytes specifies the command, the command is then stored and the buffer reset
		- then 1 byte is read which gives information on the total length in bytes
		- then, we fill the remainder of the buffer which provides the arguments for the command
```